src/coverlet.core/Helpers/InstrumentationHelper.cs

```diff
@@ -76,34 +78,86 @@ public static void DeleteHitsFile(string path)
             // Retry hitting the hits file - retry up to 10 times, since the file could be locked
             // See: https://github.com/tonerdo/coverlet/issues/25
             var retryStrategy = CreateRetryStrategy();
-
             RetryHelper.Retry(() => File.Delete(path), retryStrategy, 10);
         }
 
-        public static IEnumerable<string> GetExcludedFiles(IEnumerable<string> excludeRules,
-                                                           string parentDir = null)
+        public static bool IsModuleExcluded(string module, string[] filters)
+        {
+            if (filters == null)
+                return false;
+
+            bool isMatch = false;
+            module = Path.GetFileNameWithoutExtension(module);
+
+            foreach (var filter in filters)
+            {
+                if (!IsValidFilterExpression(filter))
```

Comment:
A lot of unnecessary calls to IsValidFilterExpression. Make sure that the input parameter filters contains valid data.